Stop sending `:event/*-update` events on card/dash pin

johnswanson opened the pull request:
When a card or dashboard is pinned, the entity itself is not changing, so let's not send the `:event/card-update` or
`:event/dashboard-update` events in these cases.

This will fix https://github.com/metabase/metabase/issues/20244

noahmoss reviewed (approved):
LGTM - definitely don't need to send a card update event when changing collection position—slightly different from archiving in that regard IMO (less destructive).

johnswanson committed:
Skip `:event/{card,dashboard}-update` on pin/unpin

When a card or dashboard is pinned, the entity itself is not changing, so let's not send the `:event/card-update` or
`:event/dashboard-update` events in these cases.
